src/Compilers/CSharp/Test/Semantic/Semantics/ReadOnlyStructsTests.cs

```diff
@@ -1000,6 +1000,78 @@ public struct S
             Assert.True(method.IsEffectivelyReadOnly);
         }
 
+        [Fact]
+        public void ReadOnlyMembers_SemanticModel()
+        {
+            var csharp = @"
+using System;
+
+public struct S1
+{
+    public void M1() {}
+    public readonly void M2() {}
+
+    public int P1 { get; set; }
+    public readonly int P2 => 42;
+    public int P3 { readonly get => 123; }
+    public int P4 { readonly set {} }
+    public static int P5 { get; set; }
+    public readonly event Action<EventArgs> E { add {} remove {} }
+}
```

Comment:
Would be nice to have a test showing
`public readonly ref readonly int P { get; }`
Which is a property that is readonly in three different ways.

Reply:
Done